magpie/api/exception.py

```diff
@@ -87,193 +86,190 @@ def verify_param(   # noqa: E126
-        if not isinstance(notEqual, bool):
-            raise TypeError("'notEqual' is not a 'bool'")
-        if not isinstance(isTrue, bool):
-            raise TypeError("'isTrue' is not a 'bool'")
-        if not isinstance(isFalse, bool):
-            raise TypeError("'isFalse' is not a 'bool'")
-        if not isinstance(isNone, bool):
-            raise TypeError("'isNone' is not a 'bool'")
-        if not isinstance(isEmpty, bool):
-            raise TypeError("'isEmpty' is not a 'bool'")
-        if not isinstance(isIn, bool):
-            raise TypeError("'isIn' is not a 'bool'")
-        if not isinstance(isEqual, bool):
-            raise TypeError("'isEqual' is not a 'bool'")
-        if not isinstance(ofType, bool):
-            raise TypeError("'ofType' is not a 'bool'")
-        if paramCompare is None and (isIn or notIn or isEqual or notEqual):
-            raise TypeError("'paramCompare' cannot be 'None' with specified test flags")
-        if isEqual or notEqual:
+        if not isinstance(not_none, bool):
+            raise TypeError("'not_none' is not a 'bool'")
+        if not isinstance(not_empty, bool):
+            raise TypeError("'not_empty' is not a 'bool'")
+        if not isinstance(not_in, bool):
+            raise TypeError("'not_in' is not a 'bool'")
+        if not isinstance(not_equal, bool):
+            raise TypeError("'not_equal' is not a 'bool'")
+        if not isinstance(is_true, bool):
+            raise TypeError("'is_true' is not a 'bool'")
+        if not isinstance(is_false, bool):
+            raise TypeError("'is_false' is not a 'bool'")
+        if not isinstance(is_none, bool):
+            raise TypeError("'is_none' is not a 'bool'")
+        if not isinstance(is_empty, bool):
+            raise TypeError("'is_empty' is not a 'bool'")
+        if not isinstance(is_in, bool):
+            raise TypeError("'is_in' is not a 'bool'")
+        if not isinstance(is_equal, bool):
+            raise TypeError("'is_equal' is not a 'bool'")
+        if not isinstance(is_type, bool):
+            raise TypeError("'is_type' is not a 'bool'")
+        if param_compare is None and (is_in or not_in or is_equal or not_equal):
+            raise TypeError("'param_compare' cannot be 'None' with specified test flags")
+        if is_equal or not_equal:
             # allow 'different' string literals for comparison, otherwise types must match exactly
-            if not (isinstance(param, six.string_types) and
-                    isinstance(paramCompare, six.string_types)) and type(param) != type(paramCompare):
-                raise TypeError("'paramCompare' cannot be of incompatible type with specified test flags")
-        if not hasattr(paramCompare, "__iter__") and (isIn or notIn):
-            paramCompare = [paramCompare]
+            if (not (isinstance(param, six.string_types) and isinstance(param_compare, six.string_types))
+                     and type(param) != type(param_compare)):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Using type() instead of isinstance() for a typecheck.](https://app.codacy.com/app/fmigneault/Magpie/pullRequest?prid=4650806)